lsp-headerline.el

```diff
@@ -144,14 +144,17 @@ caching purposes.")
   "Fix IMAGE background if it is a file otherwise return as an icon."
   (if image
       (let ((display-image (get-text-property 0 'display image)))
-        (if (listp display-image)
+        (if (and (listp display-image)
+                 (plist-member (cl-copy-list (cl-rest display-image)) :type))
```

Comment:
the cl-copy-list looks odd here.

Reply:
it's because `display-image` contains `(image :type png :file "...")`, with this we get a valid plist `(:type png :file "...")`

Reply:
how copying list helps here? It seems like we are copying it, calling plist-member, and then throwing it away.

Reply:
This line should check if a display-image contains the `type` key:
We first transform it to a plist `(cl-copy-list (cl-rest display-image))`, and then check if contains the `:type` key.
We just copy it because `cl-rest` doesn't return a `plist`

Reply:
Merging this, let me know if you think we should re-think this.